app/bundles/FormBundle/Model/FormModel.php

```diff
@@ -729,6 +731,75 @@ public function populateValuesWithGetParameters(Form $form, &$formHtml): void
         }
     }
 
+    /**
+     * Replaces strings the PHP date string parser understands with
+     * the actual time in date, datetime and hidden fields.
+     *
+     * Hidden fields must end with `|date` to prevent replacing
+     * strings by a date time representation by accident.
+     */
+    public function populateDateTimeValues(Form $form, string &$formHtml): void
+    {
+        $formName       = $form->generateFormName();
+        $fields         = $form->getFields();
+        $autoFillFields = [];
+
+        /** @var Field $field */
+        foreach ($fields as $key => $field) {
+            if (in_array($field->getType(), ['date', 'datetime', 'hidden'])) {
+                $autoFillFields[$key] = $field;
+            }
+        }
```

Comment:
A shorter syntax would be:
```suggestion
        $autoFillFields = array_filter($fields, fn($field) => in_array($field->getType(), ['date', 'datetime', 'hidden']));
```
No need to change, just food for thought.

Reply:
:D Just before submitting the PR I saw this and thought about changing it to array_filter. Then I was to lazy...